packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js

```diff
@@ -154,7 +155,28 @@ function FontCollection( { id } ) {
 	};
 
 	const handleInstall = async () => {
-		const response = await installFont( fontsToInstall[ 0 ] );
+		const fontFamily = fontsToInstall[ 0 ];
+
+		try {
+			fontFamily?.fontFace?.forEach( async ( fontFace ) => {
+				if ( fontFace.downloadFromUrl ) {
+					fontFace.file = await downloadFontFaceAsset(
+						fontFace.downloadFromUrl
+					);
+					delete fontFace.downloadFromUrl;
+				}
+			} );
```

Comment:
I don't think `forEach` [waits for promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#description), in this case I think only errors thrown by the first `fontFace` will be caught. It might be better to use something like `Promise.all`:

```suggestion
			if ( fontFamily?.fontFace ) {
				await Promise.all(
					fontFamily.fontFace.map( async ( fontFace ) => {
						if ( fontFace.downloadFromUrl ) {
							fontFace.file = await downloadFontFaceAsset(
								fontFace.downloadFromUrl
							);
							delete fontFace.downloadFromUrl;
						}
					} )
				);
			}
```

Reply:
Good catch, fixed here: [0b99a3a](https://github.com/WordPress/gutenberg/pull/57694/commits/0b99a3adf19e97ee95f624f88dbc59a64960bc61), and verified that if a font other than the first one fails to download, an error is still thrown.